gunicorn/util.py

```diff
@@ -232,33 +232,30 @@ def is_ipv6(addr):
     return True
 
 
-def parse_address(netloc, default_port=8000):
+def parse_address(netloc, default_port='8000'):
+
     if re.match(r'unix:(//)?', netloc):
         return re.split(r'unix:(//)?', netloc)[-1]
 
     if netloc.startswith("tcp://"):
         netloc = netloc.split("tcp://")[1]
 
-    # get host
+    host, port = [netloc, default_port]
+
     if '[' in netloc and ']' in netloc:
-        host = netloc.split(']')[0][1:].lower()
+        host = netloc.split(']')[0][1:]
+        port = (netloc.split(']:') + [default_port])[1]
     elif ':' in netloc:
-        host = netloc.split(':')[0].lower()
+        host, port = (netloc.split(':') + [default_port])[:2]
     elif netloc == "":
-        host = "0.0.0.0"
-    else:
-        host = netloc.lower()
-
-    #get port
-    netloc = netloc.split(']')[-1]
-    if ":" in netloc:
-        port = netloc.split(':', 1)[1]
-        if not port.isdigit():
-            raise RuntimeError("%r is not a valid port number." % port)
+        host, port = ["0.0.0.0", default_port]
+
+    try:
         port = int(port)
-    else:
-        port = default_port
-    return (host, port)
+    except ValueError:
+        raise RuntimeError("%r is not a valid port number." % port)
```

Comment:
Since master now targets Gunicorn 20, perhaps we can raise `ValueError` as it's more correct exception to raise. @benoitc @tilgovi what do you think?

```py
except ValueError:
    raise ValueError("%r is not a valid port number." % port) from None
```

Reply:
Sounds good to me!